src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/BulkPricesService.kt

```diff
@@ -0,0 +1,48 @@
+package uk.gov.justice.digital.hmpps.pecs.jpc.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.pecs.jpc.config.TimeSource
+import uk.gov.justice.digital.hmpps.pecs.jpc.price.PriceRepository
+import uk.gov.justice.digital.hmpps.pecs.jpc.price.Supplier
+import uk.gov.justice.digital.hmpps.pecs.jpc.price.effectiveYearForDate
+import uk.gov.justice.digital.hmpps.pecs.jpc.price.nextEffectiveYearForDate
+import java.util.UUID
+
+@Service
+@Transactional
+/**
+ * Service to handle bulk price related operations.
+ */
+class BulkPricesService(
+  private val priceRepository: PriceRepository,
+  private val timeSource: TimeSource
+) {
+
+  private val logger = LoggerFactory.getLogger(javaClass)
+
+  // TODO this needs to be audited when the work/code is available.
+  fun addNextYearsPrices(supplier: Supplier, multiplier: Double) {
+    val now = timeSource.date()
+
+    logger.info("Adding $supplier prices for effective year ${nextEffectiveYearForDate(now)} with multiplier $multiplier")
+
+    priceRepository.deleteBySupplierAndEffectiveYear(supplier, nextEffectiveYearForDate(now))
+
+    priceRepository.flush()
+
+    val total = priceRepository.findBySupplierAndEffectiveYear(supplier, effectiveYearForDate(now)).map {
```

Comment:
This essential does the following taken from [Confluence](https://dsdmoj.atlassian.net/wiki/spaces/PECS4/pages/2649948327/How+to+update+JPC+effective+date) page:

```
insert into prices(price_id, added_at, price_in_pence, supplier, 
from_location_id, to_location_id, effective_year)
  select uuid_generate_v4(), now(), price_in_pence * 2 as price_in_pence,
  supplier, from_location_id, to_location_id, 2021
    from prices
    where effective_year = 2020
```